nx/lib/nx/type.ex

```diff
@@ -171,6 +171,30 @@ defmodule Nx.Type do
   def to_floating({:c, size}), do: {:c, size}
   def to_floating(type), do: merge(type, {:f, 32})
 
+  @doc """
+  Converts the given type to a complex representation with
+  the minimum size necessary.
+
+  ## Examples
+
+      iex> Nx.Type.to_complex({:s, 8})
+      {:c, 64}
+      iex> Nx.Type.to_complex({:s, 32})
+      {:c, 64}
+      iex> Nx.Type.to_complex({:bf, 16})
+      {:c, 64}
+      iex> Nx.Type.to_complex({:f, 32})
+      {:c, 64}
+      iex> Nx.Type.to_complex({:s, 64})
+      {:c, 128}
+      iex> Nx.Type.to_complex({:f, 64})
+      {:c, 128}
+
+  """
+  def to_complex({:c, size}), do: {:c, size}
+  def to_complex({_name, size}) when size <= 32, do: {:c, 64}
+  def to_complex(_type), do: {:c, 128}
```

Comment:
We only need to promote f64 to c128, everything else fits in c64 if it's not already complex